db/warehouse/migrate/20230502175218_add_custom_client_name.rb

```diff
@@ -0,0 +1,23 @@
+class AddCustomClientName < ActiveRecord::Migration[6.1]
+  def change
+    create_table :CustomClientName do |t|
+      t.string :first
+      t.string :middle
+      t.string :last
+      t.string :suffix
+      t.string :use
+      t.text :notes
+      t.boolean :primary
+      t.integer :NameDataQuality
+      t.string :CustomClientNameID, null: false
+      t.string :PersonalID, null: false
+      t.string :UserID, limit: 32, null: false
+      t.integer :data_source_id
+      t.datetime :DateCreated, null: false
+      t.datetime :DateUpdated, null: false
+      t.datetime :DateDeleted
+    end
+
+    add_index :CustomClientName, [:primary, :PersonalID, :data_source_id], unique: true, name: 'unique_index_ensuring_one_primary_per_client'
```

Comment:
This should put the most diverse field first, so probably:
```suggestion
    add_index :CustomClientName, [:PersonalID, :data_source_id, :primary], unique: true, name: 'unique_index_ensuring_one_primary_per_client'
```

Reply:
@ZachCMP do you ever need this index without all three items?  If not, then we should just commit this, if you do, we should probably setup a couple of indexes.

Reply:
@eanders This index was called for in the [PT story](https://www.pivotaltracker.com/story/show/185042652). I believe this is to ensure that there's only one primary name per client ID per data source.

Reply:
Great, thanks!